File: 0463-island-perimeter/0463-island-perimeter.cpp
class Solution {
public:
 int islandPerimeter(vector<vector<int>>& grid) {
	// counter = 0;
	// DFS(grid, 0, 0);
	// return (counter);
	int gridSize = grid.size(), counter = 0;
	for (int i = 0; i < gridSize; i++)
	{
		int size = grid[i].size();
		
		for (int j = 0; j < size; j++)
		{
            int value = 4;
			if (grid[i][j] == 1)
			{
				if (i < gridSize - 1 && grid[i + 1][j] == 1)
					value--;
				if (i > 0 && grid[i - 1][j] == 1)
					value--;
				if (j < size - 1 && grid[i][j + 1] == 1)
					value--;
				if (j > 0 && grid[i][j - 1] == 1)
					value--;
                counter += value;
			}
			
		}
	}
	return (counter);
}
};

File: 0463-island-perimeter/0463-island-perimeter.py
class Solution:
    def recursive(self, grid, i, j):
        if (i < 0 or j < 0 or i > len(grid) - 1 or j > len(grid[i]) - 1 or grid[i][j] == 0):
            return 1;
        if (grid[i][j] == -1):
            return (0);
        grid[i][j] = -1;
        return self.recursive(grid, i + 1, j) + self.recursive(grid, i - 1, j) + self.recursive(grid, i, j + 1) + self.recursive(grid, i, j - 1);
    def islandPerimeter(self, grid):
        count = 0;
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                if (grid[i][j] == 1):
                    count = self.recursive(grid, i, j);
        return count;